src/api/trace/span_context.rs

```diff
@@ -22,24 +22,50 @@ pub const TRACE_FLAG_SAMPLED: u8 = TRACE_FLAGS_BIT_MASK_SAMPLED;
 /// Useful for extracting trace context
 pub const TRACE_FLAGS_UNUSED: u8 = TRACE_FLAGS_BIT_MASK_UNUSED;
 
+/// TraceId is an 16-byte value which uniquely identifies a given trace
+/// The actual `u128` value is wrapped in a tuple struct in order to leverage the newtype pattern
+#[cfg_attr(feature = "serialize", derive(Deserialize, Serialize))]
+#[derive(Clone, Debug, PartialEq, Eq, Copy, Hash)]
+pub struct TraceId(pub u128);
```

Comment:
Made the field private and implemented `impl From<u64> for SpanId`, but ran into issues unless I implemented `impl From<SpanId> for u64` on it as well. I'm aware these traits are meant to be reciprocal (i.e. defining `From<U> T` implies `Into<T> U`), but places like [here](https://github.com/open-telemetry/opentelemetry-rust/blob/4df171453503514841d1d2eca502c374394de3f0/src/api/trace/b3_propagator.rs#L139) wouldn't recognize the conversion (since it's into a generic there). 

Any advice on how to approach this, or is it fine with the double `From` impl (sorry if this is a basic question)?

Reply:
Hm yeah good point that brings up other ergonomic issues. Might be simplest to just have them be methods? e.g. something like

```rust
impl TraceId {
    pub fn from_u128(u: u128) -> Self {
        TraceId(u)
    }

    pub fn into_u128(&self) -> u128 {
        self.0
    }
}
```

Could see if that solves both problems and removes the need for explicit casting at the call sites.

Reply:
changed 👍